impeller/renderer/backend/vulkan/test/swapchain_unittests.cc

```diff
@@ -52,5 +53,43 @@ TEST(SwapchainTest, RecreateSwapchainWhenSizeChanges) {
   EXPECT_EQ(image_b->GetSize(), expected_size);
 }
 
+TEST(SwapchainTest, CachesRenderPassOnSwapchainImage) {
+  auto const context = MockVulkanContextBuilder().Build();
+
+  auto surface = CreateSurface(*context);
+  auto swapchain =
+      SwapchainVK::Create(context, std::move(surface), ISize{1, 1});
+
+  EXPECT_TRUE(swapchain->IsValid());
+
+  // We should create 3 swapchain images with the current mock setup. However,
```

Comment:
Is this enforced in the test somehow or an implied assumption?

Reply:
This can actually be fixed now. There were some global state problems, but @jason-simmons fixed up parts of the mock vulkan when he was doing the ASAN changes that make this easier to implemented.

Updated the test so we create all 3 swapchain images and can make assertions on them.